atsign-foundation/at_server
# fix: sync bug on expired keys

**- What I did**
- fixed a bug in sync where expired keys are not removed from cache resulting in multiple delete operations on commit log
**- How I did it**
- put method in hive_keystore.dart was adding key without encoding to metadata cache. Fixed the code to add encoded keys to cache
- modified logic to delete from cache - handled both encoded and actual keys
- added unit tests to group "Verify metadata cache" in hive_keystore_impl_test
**- How to verify it**
- run the group of tests "Verify metadata cache" in hive_keystore_impl_test
Fixes a bug encountered during testing of https://github.com/atsign-foundation/at_client_sdk/issues/742
